org-code-javabuilder/protocol/src/main/java/org/code/protocol/ContentManager.java

```diff
@@ -1,14 +1,48 @@
 package org.code.protocol;
 
+import java.io.FileNotFoundException;
+
 /**
  * Manages content (such as code sources and assets) that are used and generated during a
  * Javabuilder session.
  */
 public interface ContentManager {
-  byte[] getAssetData(String filename) throws JavabuilderException;
+  /**
+   * Generates a URL pointing to the asset file referenced by the asset file name.
+   *
+   * @param filename asset file name to generate a URL for
+   * @return generated URL
+   */
+  String generateAssetUrl(String filename);
 
+  /**
+   * Generates a URL that a client may use to upload a specific asset named by the provided file
```

Comment:
is the url a post or put? Would be good to be explicit in the comment so the contract is clear.

Reply:
It's a PUT - I'll update the comment.